### PR TITLE
change traceId to simple String type

### PR DESCRIPTION
**Pull request by Qi77Qi**
Stackdriver generates `X-Cloud-Trace-Context` header, which isn't a UUID. I'd like to use this traceId when it's available.

Example:
```
2019-08-08 20:00:43,295 INFO [ioapp-compute-0] o.h.s.m.Logger [Logger.scala:29] HTTP/1.1 POST /objects/lock Headers(Host: 127.0.0.1:8080, Origin: https://leonardo.dsde-dev.broadinstitute.org, User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3770.142 Safari/537.36, Accept: */*, Referer: https://leonardo.dsde-dev.broadinstitute.org/notebooks/aou-rw-test-3b102bbe/all-of-us-2278/notebooks/workspaces/qitest/qi-test.ipynb, Accept-Encoding: gzip, deflate, br, Accept-Language: en-US, en;q=0.9, zh-CN;q=0.8, zh;q=0.7, Cookie: <REDACTED>, X-Cloud-Trace-Context: 906e0fc6f601cdad3641d4fff58c6d5f/14103256683195023100, Via: 1.1 google, X-Forwarded-For: 69.173.127.109, 35.241.41.169, 35.191.2.128, 130.211.230.178, X-Forwarded-Proto: https, X-Forwarded-Host: leonardo.dsde-dev.broadinstitute.org, app:8080, X-Forwarded-Server: leonardo.dsde-dev.broadinstitute.org, all-of-us-2278-m.c.aou-rw-test-3b102bbe.internal, Content-Type: text/plain; charset=UTF-8, Connection: close, Content-Length: 47) body="{"localPath":"workspaces/qitest/qi-test.ipynb"}"
```

I think welder is the only app that uses traceId, so I'm inclined to not bump version. But I'm happy to bump it if people prefer.

Blocking https://github.com/DataBiosphere/welder/pull/43

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

**Review by rtitle — CHANGES_REQUESTED** [review]
Since it's a non-backwards-compatible change I think I'd feel better about bumping the version (even if we think only welder is using it). Otherwise 👍

File: notifications/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 This file documents changes to the `workbench-notifications` library, including notes on how to upgrade to new versions.
 
+## 0.4
+
+- upgrade `org.broadinstitute.dsde.workbench.model`
+
+SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-util" % "0.4-TRAVIS-REPLACE-ME"`
```
Comment (rtitle) [inline notifications/CHANGELOG.md]: Should say workbench-notifications?

File: google2/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,12 @@
 
 This file documents changes to the `workbench-google2` library, including notes on how to upgrade to new versions.
 
+## 0.6
+
+- upgrade `org.broadinstitute.dsde.workbench.model`
+
+SBT dependency: `"org.broadinstitute.dsde.workbench" %% "workbench-util" % "0.6-TRAVIS-REPLACE-ME"`
```
Comment (rtitle) [inline google2/CHANGELOG.md]: Should say `workbench-google2`

**Review by rtitle — CHANGES_REQUESTED** [review]
A couple typos in changelogs.. also I did not see CHANGELOG updates for workbench-google and workbench-service-test.

Is it standard practice to bump _all_ modules if making changes to `workbench-model`? I don't think I've done that in the past.

**Comment by Qi77Qi** [comment]
@rtitle in the past, I've always bumped all modules when dependency changed, which I think makes sense, because it breaks those modules bin compatibility(which is why I didn't feel like bumping model in the beginning).

**Review by rtitle — APPROVED** [review]

File: model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala [hunk 3]
```diff
@@ -1,6 +1,4 @@
 package org.broadinstitute.dsde.workbench.model
 
-import java.util.UUID
-
-// uuid for tracing a unique call flow in logging
-final case class TraceId(uuid: UUID) extends AnyVal
+// unique identifier for tracing a unique call flow in logging
+final case class TraceId(asString: String) extends AnyVal
```
Comment (dvoet) [inline model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala:6]: you could provide a companion object with an apply function that takes a UUID and remain backwards compatible.

Comment (Qi77Qi) [reply model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala:6]: it feels to me too much effort and making code more complicated than necessary given no one else but me has used this field in welder repo

Comment (dvoet) [reply model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala:6]: This repo is special. It is used by many services. We need to go through extra pains to be backwards compatible. The first rule of making breaking api changes is don't (see https://github.com/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md).

Comment (Qi77Qi) [reply model/src/main/scala/org/broadinstitute/dsde/workbench/model/models.scala:6]: @dvoet , updated. Let me know if you have more comments

**Comment by Qi77Qi** [comment]
@dvoet do you have more comments on this?

**Review by dvoet — APPROVED** [review]

**Pull request merged**